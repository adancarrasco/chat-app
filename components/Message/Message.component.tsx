import React, { useState } from "react";
import { StyleSheet, Text, Animated } from "react-native";

interface IMessageProps {
  isLocalUser: boolean;
  userName: string;
  message: string;
  shouldAnimate: boolean;
}

export default function Message({
  isLocalUser,
  userName,
  message,
  shouldAnimate
}: IMessageProps) {
  const {
    container,
    localUser,
    remoteUser,
    userNameText,
    messageText
  } = styles;
  const [fadeAnim] = useState(new Animated.Value(0));

  React.useEffect(() => {
    Animated.timing(fadeAnim, {
      toValue: 1,
      duration: 700
    }).start();
  }, []);

  return (
    <Animated.View
      style={[
        container,
        isLocalUser ? localUser : remoteUser,
        shouldAnimate ? { opacity: fadeAnim } : ""
      ]}
    >
      <Text style={userNameText}>{userName}</Text>
      <Text style={messageText}>{message}</Text>
    </Animated.View>
  );
}

const styles = StyleSheet.create({
  container: {
    flexDirection: "column",
    padding: 10,
    borderRadius: 8,
    color: "#fff",
    marginBottom: 14,
    maxWidth: "65%"
  },
  localUser: {
    backgroundColor: "rgb(140,149,154)"
  },
  remoteUser: {
    backgroundColor: "rgb(43,114,221)",
    alignItems: "flex-end",
    color: "#fff",
    alignSelf: "flex-end"
  },
  userNameText: {
    color: "#fff",
    marginBottom: 6,
    fontWeight: "600"
  },
  messageText: {
    color: "#fff"
  }
});
