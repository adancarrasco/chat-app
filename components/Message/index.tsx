export { default } from "./Message.component";
