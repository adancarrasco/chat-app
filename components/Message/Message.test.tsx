import React from "react";
import renderer from "react-test-renderer";
import Message from "./";

describe("<Message /> rendering", () => {
  const mockMessage = {
    isLocalUser: true,
    userName: "Summer",
    message: "Sorry I cannot call you now.",
    shouldAnimate: false
  };

  it("Matches snapshot as localUser", () => {
    const tree = renderer.create(<Message {...mockMessage} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Matches snapshot as remoteUser", () => {
    const tree = renderer
      .create(<Message {...mockMessage} isLocalUser={false} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
