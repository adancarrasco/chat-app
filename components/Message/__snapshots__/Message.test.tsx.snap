// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Message /> rendering Matches snapshot as localUser 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgb(140,149,154)",
      "borderRadius": 8,
      "color": "#fff",
      "flexDirection": "column",
      "marginBottom": 14,
      "maxWidth": "65%",
      "padding": 10,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#fff",
        "fontWeight": "600",
        "marginBottom": 6,
      }
    }
  >
    Summer
  </Text>
  <Text
    style={
      Object {
        "color": "#fff",
      }
    }
  >
    Sorry I cannot call you now.
  </Text>
</View>
`;

exports[`<Message /> rendering Matches snapshot as remoteUser 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-end",
      "alignSelf": "flex-end",
      "backgroundColor": "rgb(43,114,221)",
      "borderRadius": 8,
      "color": "#fff",
      "flexDirection": "column",
      "marginBottom": 14,
      "maxWidth": "65%",
      "padding": 10,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#fff",
        "fontWeight": "600",
        "marginBottom": 6,
      }
    }
  >
    Summer
  </Text>
  <Text
    style={
      Object {
        "color": "#fff",
      }
    }
  >
    Sorry I cannot call you now.
  </Text>
</View>
`;
